aws/cloudformation-templates/base/acm-cert.yaml

```diff
@@ -0,0 +1,226 @@
+                        acmarn_parameter = ssm_client.put_parameter(
+                            Name=acmarn_param_name,
+                            Value=my_response['CertificateArn'],
+                            Type='String',
+                            Overwrite=True)
+                            
+                        response_data['certificate_arn'] = my_response['CertificateArn']
+                        response_data['Message'] = "Resource creation succeeded"
+                    elif event['RequestType'] == 'Update':
+                        response_data['Message'] = "Resource update succeeded"
+                    elif event['RequestType'] == 'Delete':
+                        # Delete the cert from ACM, assumes all attachments are already removed.
+                        
+                        # Retrieve ACM ARN from Parameter store
+                        acmarn_parameter = ssm_client.get_parameter(Name=acmarn_param_name)
+                        
+                        # Delete ACM
+                        my_response = acm_client.delete_certificate(
+                            CertificateArn=acmarn_parameter['Parameter']['Value']
+                        )
+                        
+                        response_data['Message'] = "Resource deletion succeeded"                      
+                except ClientError as e:
+                  print("Error: " + str(e))
+                  response_status = cfnresponse.FAILED
+                  response_data['Message'] = "Resource {} failed: {}".format(event['RequestType'], e)
+
+                cfnresponse.send(event, context, response_status, response_data)
+          Handler: index.handler            
+          Runtime: python3.9
+          Timeout: 120
+          Role: !GetAtt ACMimportCertLambdaExecutionRole.Arn
+          Layers:
+            - !Ref ACMimportCertLambdaLambdaLayer1
+        
+    ACMimportCertLambdaExecutionRole:
+        Type: 'AWS::IAM::Role'
+        Properties:
+          AssumeRolePolicyDocument:
+            Version: 2012-10-17
+            Statement:
+            - Effect: Allow
+              Principal:
+                Service:
+                  - lambda.amazonaws.com
+              Action:
+                - 'sts:AssumeRole'    
+          Path: /
+          Policies:
+              - PolicyName: CustomLogAndSSMPolicy
+                PolicyDocument:
+                  Version: 2012-10-17
+                  Statement:
+                  - Effect: Allow
+                    Action:
+                      - logs:CreateLogStream
+                      - logs:DescribeLogStreams
+                      - logs:PutLogEvents
+                    Resource:
+                      - !Sub 'arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:*'
+                  - Effect: Allow
+                    Action:
+                      - ssm:GetParameter
+                      - ssm:PutParameter
+                    Resource: '*'
```

Comment:
Lock this down the parameter being read/updated.

Reply:
> Where is the lambda layer code? How will it be maintained?
Lambda layer code is already committed under src/aws-lambda/acm-cert-lambda-layer/lambda-layer.zip

> Lock this down the parameter being read/updated.
Done - https://github.com/aws-samples/retail-demo-store/pull/369/commits/b2f8d93d6f166be5d34f08c571c2b124966d7868